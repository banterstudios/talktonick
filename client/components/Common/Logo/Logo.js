import React, { PureComponent } from 'react'
import glamorous from 'glamorous'
import LazyImage from 'client/components/Gui/LazyImage'
import { logo } from 'client/consts/images'
import PropTypes from 'prop-types'
import { checkIfValueIsAllowedOrSetDefault } from 'client/utils/objectUtils'

const allowedModifiers = ['sm', 'md', 'lg']

const LogoWrapper = glamorous.div(({ modifier, theme: { logoSmWidth, logoSmHeight} }) => ({
  position: 'relative',
  width: modifier === 'sm' ? logoSmWidth : '20px',
  height: modifier === 'sm' ? logoSmHeight : '20px'
}))

export default class Logo extends PureComponent {
  static propTypes = {
    modifier: PropTypes.string
  }

  render () {
    const { modifier } = this.props

    return (
      <LogoWrapper modifier={checkIfValueIsAllowedOrSetDefault(modifier, allowedModifiers, 'sm')}>
        <LazyImage src={logo} />
      </LogoWrapper>
    )
  }
}
