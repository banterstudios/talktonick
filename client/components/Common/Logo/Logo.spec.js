import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Logo from './Logo'

describe('components/Common/Logo', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = shallow(<Logo />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })
})
