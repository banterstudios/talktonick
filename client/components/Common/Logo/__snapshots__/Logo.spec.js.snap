// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/Logo @renders should render correctly 1`] = `
<glamorous(div)
  modifier="sm"
>
  <Link
    href="/"
    linkType="route"
    target="_blank"
  >
    <LazyImage
      bgPos="center"
      bgRepeat="no-repeat"
      bgSize="cover"
      src="/static/assets/images/logo.png"
      useBgImage={false}
    />
  </Link>
</glamorous(div)>
`;
